app/javascript/components/artists/UpdateArtist.jsx

```diff
@@ -3,22 +3,47 @@ import React from 'react';
 import { Button, Dialog, Intent } from "@blueprintjs/core";
 
 
-
 class UpdateArtist extends React.Component {
   constructor(props) {
     super(props);
+    const { name, media, program, description, avatar, id, featured_work_id } = this.props.artist;
     this.state = {
       artist: {
-        name: this.props.artist.name,
-        genres: this.props.artist.genres,
-        program: this.props.artist.program,
-        description: this.props.artist.description,
-        avatar: this.props.artist.avatar,
-        artist_id: this.props.artist.id
-      }
+        name,
+        media,
+        program,
+        description,
+        avatar,
+        featured_work_id,
+        artist_id: id
+      },
+      works: [],
+      categories: {},
+      componentDidMount: false,
     }
   }
 
+  componentDidMount = () => {
+    const works_route = APIRoutes.artists.works(this.props.artist.id);
+    const categories_route = APIRoutes.artists.categories;
+    Promise.all([
+      Requester.get(works_route),
+      Requester.get(categories_route)
+    ]).then(
+      response => {
+        const [works_response, categories_response] = response;
+        this.setState({
+          works: works_response,
+          categories: categories_response,
+          componentDidMount: true
+        });
+      },
+      error => {
```

Comment:
Probably good to add a `FIXME` comment here, or create a tech debt issue about this - errors here will leave your component in an indefinite loading state